pennylane/fermi/conversion.py

```diff
@@ -148,3 +148,145 @@ def _(fermi_operator: FermiSentence, ps=False, wire_map=None, tol=None):
         return qubit_operator.map_wires(wire_map)
 
     return qubit_operator
+
+
+def parity_transform(
+    fermi_operator: (Union[FermiWord, FermiSentence]),
+    n_qubits: int,
+    ps: bool = False,
+    wire_map: dict = None,
+    tol: float = None,
+) -> Union[Operator, PauliSentence]:
+    r"""Convert a fermionic operator to a qubit operator using the parity mapping.
+    In parity mapping, qubit j stores the parity of all j-1 qubits before it whereas in Jordan-Wigner, qubit j stores the occupation number of qubit j.
+
+    The fermionic creation and annihilation operators are mapped to the Pauli operators as
+
+    .. math::
+        a^{\dagger}_0 =  \left (\frac{X_0 - iY_0}{2}  \right )\otimes X_1 \otimes X_2 \otimes ... X_N, \:\: \text{...,} \:\:
+
+        a^{\dagger}_n = \left (\frac{Z_{n-1} \otimes X_n - iY_n}{2} \right ) \otimes X_{n+1} \otimes X_{n+2} \otimes ... \otimes X_N
+
+    and
+
+    .. math::
+
+        a_0 =  \left (\frac{X_0 + iY_0}{2}  \right )\otimes X_1 \otimes X_2 \otimes ... X_N, \:\: \text{...,} \:\:
+
+        a_n = \left (\frac{Z_{n-1} \otimes X_n - iY_n}{2} \right ) \otimes X_{n+1} \otimes X_{n+2} \otimes ... \otimes X_N
+
+    where :math:`X`, :math:`Y`, and :math:`Z` are the Pauli operators and N is the number of qubits/spin orbitals.
+
+    Args:
+        fermi_operator(FermiWord, FermiSentence): the fermionic operator
+        ps (bool): whether to return the result as a PauliSentence instead of an
+            Operator. Defaults to False.
+        wire_map (dict): a dictionary defining how to map the oribitals of
+            the Fermi operator to qubit wires. If None, the integers used to
+            order the orbitals will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        Union[PauliSentence, Operator]: a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> parity_transform(w, n_qubits=6)
+    (-0.25j*(PauliY(wires=[0]))) + ((-0.25+0j)*(PauliX(wires=[0]) @ PauliZ(wires=[1]))) +
+    ((0.25+0j)*(PauliX(wires=[0]))) + (0.25j*(PauliY(wires=[0]) @ PauliZ(wires=[1])))
+
+    >>> parity_transform(w, n_qubits=6, ps=True)
+    -0.25j * Y(0)
+    + (-0.25+0j) * X(0) @ Z(1)
+    + (0.25+0j) * X(0)
+    + 0.25j * Y(0) @ Z(1)
+
+    >>> parity_transform(w, n_qubits=6, ps=True, wire_map={0: 2, 1: 3})
+    -0.25j * Y(2)
+    + (-0.25+0j) * X(2) @ Z(3)
+    + (0.25+0j) * X(2)
+    + 0.25j * Y(2) @ Z(3)
+    """
+    return _parity_transform_dispatch(fermi_operator, n_qubits, ps, wire_map, tol)
+
+
+@singledispatch
```

Comment:
I chose to use singledispatch as it is a clean and convenient way to implement functions that behave differently based on the first argument in the function. In other words, it is a way to achieve some form of function overloading in python. Here, I wanted to implement this function with both fermiword and fermisentence and also make sure that it throws an error when the wrong input type is used. This also fits well with the overall structure of the previously written code that I was developing on top of.
This can be done using object oriented programming by defining multiple classes for each argument type. In this case, it would be ideal to change the structure of jordan-wigner mapping implemented before this. 
Another way to do this is to check the type of the argument and implement different behaviors using if/else statements. This makes the code hard to read.

Reply:
Awesome, that makes perfect sense 👍🏼